pkgs/build-support/fetchsteam/default.nix

```
@@ -0,0 +1,56 @@
+{
+  lib,
+  runCommand,
+  depotdownloader,
+  cacert,
+  writeText,
+}:
+
+{
+  name ? "${toString app}-${toString depot}-${toString manifest}-depot",
+  app,
+  depot,
+  manifest,
+  branch ? null,
+  language ? null,
+  lowViolence ? false,
+  fileList ? [ ],
+  fileListRegex ? false,
+  debug ? false,
+  hash ? lib.fakeHash,
+}:
+
+runCommand name
+  {
+    nativeBuildInputs = [ depotdownloader ];
+
+    env.SSL_CERT_FILE = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+
+    outputHash = hash;
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+  }
+  ''
+    # Hack to prevent DepotDownloader from crashing trying to write to
+    # ~/.local/share/
+    export HOME
+    HOME=$(mktemp -d)
+
+    DepotDownloader \
+      -app "${toString app}" \
+      -depot "${toString depot}" \
+      -manifest "${toString manifest}" \
+      ${lib.optionalString (branch != null) "-beta ${branch}"} \
+      ${lib.optionalString (language != null) "-language ${language}"} \
+      ${lib.optionalString lowViolence "-lowviolence"} \
+      ${
+        lib.optionalString (fileList != [ ]) (
+          (lib.optionalString fileListRegex "regex:")
+          + (writeText "steam-file-list-${name}.txt" (lib.concatStringsSep "\n" fileList))
+        )
+      } \
+      ${lib.optionalString debug "-debug"} \
+      -loginid ${toString (manifest - (manifest / 4294967295) * 4294967295)} \
```

```suggestion
      -loginid ${toString (manifest - (manifest / 4294967295) * 4294967295)} \
```
There should probably be a comment as to why this is done?

---

I agree, but the problem is: where do I put that comment?